.github/workflows/build-docker.yaml

```
@@ -0,0 +1,43 @@
+name: Build and push spellcheck image
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: alexslemonade/spellcheck
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker login
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=semver,pattern={{raw}}
+            type=edge,branch=main
+
+      - name: Build and push to Docker
+        uses: docker/build-push-action@v5
+        with:
+          push: true
```

Push only if this is a push to main.
```suggestion
          push: ${{ github.event.action == 'push' }
```

---

ah, `}`